lib/sequent/core/command_service.rb

```diff
@@ -28,22 +27,24 @@ def initialize
     # * Unit of Work is cleared
     #
     class CommandService
-
       class << self
-        attr_accessor :configuration,
-                      :instance
-      end
+        def configure
+          configuration = instance.configuration.dup
+          yield(configuration) if block_given?
+          @instance = new(configuration) # this makes it threadsafe
```

Comment:
Does Ruby (the language, not the MRI) guarantee thread-safety of instance variable assignment? Also, when multiple threads are configuring concurrently some configuration will be lost (last thread wins?).

Actually, I don't think thread-safety is a real concern here (all handlers are probably registered at load time in a single thread), but the comment might be misleading.